src/plugins/walletConnectToDapps/WalletConnectBridgeProvider.tsx

```diff
@@ -0,0 +1,93 @@
+import { HDWalletWCBridge } from '@shapeshiftoss/hdwallet-walletconnect-bridge'
+import type { WalletConnectCallRequest } from '@shapeshiftoss/hdwallet-walletconnect-bridge/dist/types'
+import type { FC, PropsWithChildren } from 'react'
+import { useCallback, useEffect, useState } from 'react'
+import { useWallet } from 'hooks/useWallet/useWallet'
+
+import { CallRequestModal } from './components/modal/callRequest/CallRequestModal'
+import { WalletConnectBridgeContext } from './WalletConnectBridgeContext'
+
+export const WalletConnectBridgeProvider: FC<PropsWithChildren> = ({ children }) => {
+  const wallet = useWallet().state.wallet
+  const [bridge, setBridge] = useState<HDWalletWCBridge>()
+
+  const [callRequests, setCallRequests] = useState<WalletConnectCallRequest[]>([])
+  const onCallRequest = useCallback(
+    (request: WalletConnectCallRequest) => setCallRequests(prev => [...prev, request]),
+    [],
+  )
+  const approveRequest = useCallback(
+    async (request: WalletConnectCallRequest, approveData: unknown) => {
+      await bridge?.approveRequest(request, approveData as any)
+      setCallRequests(prev => prev.filter(req => req.id !== request.id))
+    },
+    [bridge],
+  )
+  const rejectRequest = useCallback(
+    async (request: WalletConnectCallRequest) => {
+      await bridge?.rejectRequest(request)
+      setCallRequests(prev => prev.filter(req => req.id !== request.id))
+    },
+    [bridge],
+  )
+
+  const [, setTick] = useState(0)
+  const rerender = useCallback(() => setTick(prev => prev + 1), [])
+
+  const disconnect = useCallback(async () => {
+    await bridge?.disconnect()
+    setBridge(undefined)
+  }, [bridge])
+
+  const connect = useCallback(
+    async (uri: string) => {
+      if (!wallet) {
+        alert('TODO: No HDWallet connected')
```

Comment:
we don't even want to get to this code path - hide the walletconnect icon in the menus if a wallet is not connected